### DISCUSSION TITLE
How To Create Release

### DISCUSSION THREAD

**Discussion by 7yl4r** [iobis/pyobis#84] [Announcements]
Checklist for GitHub release creation:

1. ensure `Changelog.rst` is fully up-to-date
2. use GitHub "releases" web GUI to add a tag with version name (use semantic versioning in the form `v0.0.0`)
3. name the release with the same version number
4. use GitHub "Generate Release Notes"
5. manually add notes to the top of the release description
6. click "publish release"

**Comment by ocefpaf** [comment]
> Checklist for GitHub release creation:
> 
> 1. add a commit to update the `__version__` in `pyobis/__init__.py` (use semantic versioning)

We are using setuptools-scm so that step should not be needed. The tagged version will be written to the version.py file when creating a tarball. All we need is to tag a release on GH.

> 2. use GitHub "releases" web GUI to add a tag with version name matching the new version number
> 3. name the release with the same version number

The important part here is to have a stable versioning scheme. Our regex catches vMAJOR.MINOR.MICRO, so 1.2.5 would be v1.2.5.

> 4. use GitHub "Generate Release Notes"
> 5. manually add notes to the top of the release description
> 6. click "publish release"

:+1: I'm looking into the PyPI token issue now BTW.

**Reply by 7yl4r** [reply]
> +1 I'm looking into the PyPI token issue now BTW.

I think the token is working and using on:release did trigger as it should. The last run did not publish because of a malformed parameter ([gh actions run here](https://github.com/iobis/pyobis/runs/8178999787?check_suite_focus=true)) but I think I have now resolved that. 

Hopefully it works next time we are cutting a new release on gh. I expect we will do that sometime later this week.

**Reply by ocefpaf** [reply]
Everything is working now as expected! Thanks @7yl4r for fixing it.

https://pypi.org/project/pyobis/1.2.6/

**Reply by ayushanand18** [reply]
So good to see the PyPI release. Thank you!
